ViewControllers/DetailViewController.swift

```diff
@@ -202,40 +238,67 @@ class DetailViewController: UIViewController {
         
-        view.addSubview(titleLabel)
-        view.addSubview(imageView)
-        view.addSubview(releaseStack)
-        view.addSubview(durationStack)
-        view.addSubview(movieDescriptionLabel)
-        view.addSubview(collectionView)
+        contentView.addSubview(titleLabel)
+        contentView.addSubview(imageView)
+        contentView.addSubview(releaseStack)
+        contentView.addSubview(durationStack)
+        contentView.addSubview(movieDescriptionLabel)
+        contentView.addSubview(collectionView)
         
         NSLayoutConstraint.activate([
-            titleLabel.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor),
-            titleLabel.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
-            titleLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            titleLabel.topAnchor.constraint(equalTo: contentView.layoutMarginsGuide.topAnchor),
+            titleLabel.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
+            titleLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
 
             imageView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 16),
             imageView.centerXAnchor.constraint(equalTo: titleLabel.centerXAnchor),
-            imageView.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 0.61),
+            imageView.widthAnchor.constraint(equalTo: contentView.widthAnchor, multiplier: 0.61),
             imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 1.41),
             
-            releaseStack.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
+            releaseStack.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
             releaseStack.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 35),
 
             durationStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             durationStack.topAnchor.constraint(equalTo: releaseStack.bottomAnchor, constant: 8),
 
             movieDescriptionLabel.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            movieDescriptionLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            movieDescriptionLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
             movieDescriptionLabel.topAnchor.constraint(equalTo: durationStack.bottomAnchor, constant: 16),
 
             collectionView.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             collectionView.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
             collectionView.rightAnchor.constraint(equalTo: movieDescriptionLabel.rightAnchor),
-            collectionView.heightAnchor.constraint(equalToConstant: 200),
+            collectionView.heightAnchor.constraint(equalToConstant: calculateCollectionViewHeight())
```

Comment:
Things to keep in mind here:
- Even though there is a method to calculate collection height, this is called only once and the collection height constraint will be static. In our view here this is working because `setupUI` is called only after data is retrieved so the collection can be sized properly. Ideally `setupUI` should be called before the view is even displayed (at `loadView` method for example). In some cases our screen will have a "loading state" to be shown before content is retrieved.
- There is no constraint from collection to the bottom of the content view. For the content view to be able to know its own size, everything should be linked internally. In our case here, all subviews are properly interconnected up until the last link, from bottom of collection to bottom of content view. If we add that connection then your content view will be able to know its own size dynamically depending on its subviews sizes.

In case you want/need to have a constraint that you can update manually you can do like this:
```swift
class MyViewController: UIViewController {
    private var myHeightConstraint: NSLayoutConstraint?
    private let myView = UIView()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.addSubview(myView)
        myHeightConstraint = myView.heightAnchor.constraint(equalToConstant: 10)
        myHeightConstraint.isActive = true
    }

    private func updateHeight(to newHeight: CGFloat) {
        myHeightConstraint.constant = newHeight
    }
}
```

Reply:
I tried pinning the collectionView to the bottom of the contentView but it just failed to display, all sizes being zero